eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,225 @@
+            return;
+        }
+
+        switch (ast.getType()) {
+            case TokenTypes.VARIABLE_DEF:
+                isNameLogger(ast);
+                isNonStaticLoggerInstance(ast);
+                break;
+            case TokenTypes.METHOD_CALL:
+                isGuardConditionChecked(ast);
+                break;
+        }
+    }
+
+    /**
+     * Check if the file has Logger class imported
+     * @param rootAST The root of AST
+     * @return true if the class has Logger class imported.
+     */
+    private boolean hasLoggerImport(DetailAST rootAST) {
+        DetailAST importToken = rootAST.findFirstToken(TokenTypes.IMPORT);
+        while (importToken != null && importToken.getType() == TokenTypes.IMPORT ) {
+            String importClassName = (importToken.findFirstToken(TokenTypes.DOT)).getLastChild().getText();
+            if (importClassName.equals(LOGGER_CLASS)) {
+                return true;
+            }
+            importToken = importToken.getNextSibling();
+        }
+        return false;
+    }
+
+    /**
+     * Check if the Logger instance named as logger, log as error if not
+     * @param varDefAST the VARIABLE_DEF node
+     */
+    private void isNameLogger(DetailAST varDefAST) {
+        String type = varDefAST.findFirstToken(TokenTypes.TYPE).getFirstChild().getText();
+        if (type.equals(LOGGER_CLASS)) {
+            String varName = varDefAST.findFirstToken(TokenTypes.IDENT).getText();
+            loggerSet.add(varName);
+            if (!varName.equals(LOGGER_NAME)) {
+                log(varDefAST.getLineNo(), String.format(LOGGER_NAME_ERR, varName));
+            }
+        }
+    }
+
+    /**
+     * Check if the Logger is static instance, log as error if static instance logger,
+     * @param varDefAST the VARIABLE_DEF node
+     */
+    private void isNonStaticLoggerInstance(DetailAST varDefAST) {
+
+        String type = varDefAST.findFirstToken(TokenTypes.TYPE).getFirstChild().getText();
+        if (type.equals(LOGGER_CLASS)) {
+            if(varDefAST.findFirstToken(TokenTypes.MODIFIERS).branchContains(TokenTypes.LITERAL_STATIC)) {
+                log(varDefAST.getLineNo(), STATIC_LOGGER_ERR);
+            }
+        }
+    }
+
+    /**
+     * Always guard with conditional block whenever a log function called.
+     * @param methodCallAST The METHOD_CALL node
+     */
+    private void isGuardConditionChecked(DetailAST methodCallAST) {
+        DetailAST loggerName = methodCallAST.findFirstToken(TokenTypes.DOT).getFirstChild();
+        // only check for logger instance, skip checking otherwise
+        if (!loggerSet.contains(loggerName.getText())) {
+            return;
+        }
+
+        DetailAST methodName = methodCallAST.findFirstToken(TokenTypes.DOT).getLastChild();
+        String methodNameStr = methodName.getText();
+        if (isLogLevelMatched(methodCallAST, methodNameStr)) {
+            switch (methodNameStr) {
+                case FATAL: // log4j only
```

Comment:
I thought these were for track 2 SDKs. I don't think we're using log4j

Reply:
as we have discussed in the Java Channel.  We can't remove the log4j from track 1. So it is reasonable to consider log4j in the here.

Reply:
This is not correct. These checks are only intended to be run on track two client libraries. You should chat with Alan about only running your checks on track two client libraries.

Reply:
double check. All of the new custome check style only checks on track two client libraries?

Reply:
Yes